python/ray/train/torch/torch_checkpoint.py

```diff
@@ -78,7 +78,8 @@ def __getstate__(self) -> dict:
     def __setstate__(self, state: dict):
         if "_data_dict" in state:
             state = state.copy()
-            state["_data_dict"] = self._decode_data_dict(state["_data_dict"])
+            if state["_data_dict"]:
```

Comment:
I'd move this if to the outer if to avoid an unnecessary shallow copy

Reply:
ah right, thanks for catching this!